cilium-cli/connectivity/check/context.go

```diff
@@ -945,11 +945,21 @@ func (ct *ConnectivityTest) CurlCommand(peer TestPeer, ipFam features.IPFamily,
 		cmd = append(cmd, "-H", fmt.Sprintf("Host: %s", strings.TrimSuffix(host, ".")))
 	}
 
+	numTargets := 1
+	if ct.params.CurlParallel > 0 {
+		numTargets = int(ct.params.CurlParallel)
+		cmd = append(cmd, "--parallel", "--parallel-immediate")
+	}
+
 	cmd = append(cmd, opts...)
-	cmd = append(cmd, fmt.Sprintf("%s://%s%s",
-		peer.Scheme(),
-		net.JoinHostPort(peer.Address(ipFam), fmt.Sprint(peer.Port())),
-		peer.Path()))
+
+	for i := 0; i < numTargets; i++ {
```

Comment:
nit: could use the range-over-int feature [added in Go 1.22](https://go.dev/doc/go1.22#language) here:

```suggestion
	for i := range numTargets {
```

Reply:
TIL, thanks!

Reply:
Applied!